### PR TITLE
RUMM-2746 Sync SR touch and screen recorders

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/RecorderOnDrawListener.kt [hunk 1]
```diff
@@ -20,14 +18,14 @@ internal class RecorderOnDrawListener(
     private val pixelsDensity: Float,
     private val processor: Processor,
     private val snapshotProducer: SnapshotProducer,
-    private val handler: Handler = Handler(Looper.getMainLooper())
+    private val debouncer: Debouncer = Debouncer()
 ) : ViewTreeObserver.OnDrawListener {
     private var currentOrientation = Configuration.ORIENTATION_UNDEFINED
     private val trackedActivity: WeakReference<Activity> = WeakReference(activity)
+    private var lastScreenUpdateInNs = System.nanoTime()
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/RecorderOnDrawListener.kt:25]: seems like this field is not used: it is written, but never read.

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        assertThat(fakeRunnable.wasExecuted).isFalse
+        assertThat(fakeSecondRunnable.wasExecuted).isTrue
+    }
+
+    @Test
+    fun `M execute the runnable once W debounce { high frequency, delay threshold reached  }`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeDelayedRunnables = forge.aList(size = forge.anInt(min = 0, max = 10)) { TestRunnable() }
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt:92]: having `min=0` will make it possible to have empty `fakeDelayedRunnables` list, leading to the division by zero on the line below. Same for the test below.

Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt:92]: oh damn...yes, I actually had `min=1` there at some point I don't know why I changed it back. Nicely spotted.

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    private class TestRunnable : Runnable {
+        var wasExecuted: Boolean = false
+
+        override fun run() {
+            wasExecuted = true
+        }
+    }
+
+    companion object {
+        private val TEST_MAX_DELAY_THRESHOLD_IN_NS = TimeUnit.SECONDS.toNanos(2)
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt:166]: minor: this probably can be even lower to make tests faster

**Review by 0xnm — APPROVED** [review]

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt (hunk 3 above)
Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt:166]: yup...I hesitated as not to have longer tests, we already have quite a lot but I guess is better than to have flaky ones.

**Review by xgouchet — APPROVED** [review]

**Pull request merged**